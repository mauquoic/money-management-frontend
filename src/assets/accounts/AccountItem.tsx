import {Account} from "../../models/Account";
import React from "react";
import TableRow from "@mui/material/TableRow";
import TableCell from "@mui/material/TableCell";

type Props = {
  account: Account
}

export const AccountItem = ({account}: Props) => {
  return (
    <TableRow hover role="checkbox" tabIndex={-1}>
      <TableCell align={'center'}>
        {account.name}
      </TableCell>
      <TableCell align={'center'}>
        {account.currency} {account.amount}
      </TableCell>
      <TableCell align={'center'}>
        {account.description}
      </TableCell>
    </TableRow>
  );
};