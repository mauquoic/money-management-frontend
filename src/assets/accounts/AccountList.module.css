.root {
    min-width: 100%;
}

.container {
    max-height: 70%;
}