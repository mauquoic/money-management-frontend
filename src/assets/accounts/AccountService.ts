import {AxiosResponse} from "axios";
import {Account} from "../../models/Account";
import axiosInstance from "../../shared/axios";

class AccountService {
  static getAccounts = async () => {
    return await axiosInstance.get("https://run.mocky.io/v3/1a0f21ef-f96b-4b94-92a4-7afea097e0fb")
      .then((response: AxiosResponse<Array<Account>>) => {
        return response.data
      })
      .catch(() => {
        alert("Unable to retrieve the accounts.");
        return [];
      });
  }
}

export default AccountService;