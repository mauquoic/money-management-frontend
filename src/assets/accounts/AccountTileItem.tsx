import React from "react";
import {Account} from "../../models/Account";
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import {Grid, makeStyles} from "@material-ui/core";

type Props = {
  account: Account
}

const useStyles = makeStyles({
  root: {
    height: "100%"
  },
  title: {
    fontSize: 14,
  },
});

export const AccountTileItem = ({account}: Props) => {
  const classes = useStyles();

  return (
    <Grid item xs={12} sm={6} md={4} xl={3}>
      <Card className={classes.root} elevation={12}>
        <CardContent>
          <Typography color="textPrimary" gutterBottom variant="h5">
            {account.name}
          </Typography>
          <Typography color="textSecondary">
            {account.currency} {account.amount}
          </Typography>
          <Typography>
            {account.description}
          </Typography>
        </CardContent>
        {/*<CardActions>*/}
        {/*  <Button size="small">Edit</Button>*/}
        {/*</CardActions>*/}
      </Card>
    </Grid>
  )
};
