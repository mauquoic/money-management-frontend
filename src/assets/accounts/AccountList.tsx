import React, {useEffect, useState} from "react";
import {Account} from "../../models/Account";
import {AccountItem} from "./AccountItem";
import classes from "./AccountList.module.css";
import Table from "@mui/material/Table";
import TableHead from "@mui/material/TableHead";
import TableRow from "@mui/material/TableRow";
import TableCell from "@mui/material/TableCell";
import TableBody from "@mui/material/TableBody";
import Paper from "@mui/material/Paper";
import TableContainer from "@mui/material/TableContainer";

type Props = {
  accounts: Array<Account>
}

function AccountTableList({accounts}: Props) {

  return (
    <div className={classes.root}>
      <Paper elevation={6} className={classes.root}>
        <TableContainer className={classes.container}>
          <Table stickyHeader aria-label="sticky table">
            <TableHead>
              <TableRow>
                <TableCell align={'center'}>
                  Name
                </TableCell>
                <TableCell align={'center'}>
                  Value
                </TableCell>
                <TableCell align={'center'}>
                  Description
                </TableCell>
              </TableRow>
            </TableHead>
            <TableBody>
              {accounts?.map((account) => {
                return (
                  <AccountItem key={account.id} account={account}/>
                );
              })}
            </TableBody>
          </Table>
        </TableContainer>
      </Paper>
    </div>
  )
}

export default AccountTableList;