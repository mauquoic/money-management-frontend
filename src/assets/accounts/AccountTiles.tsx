import React from "react";
import {Account} from "../../models/Account";
import {Grid} from "@material-ui/core";
import {AccountTileItem} from "./AccountTileItem";

type Props = {
  accounts: Array<Account>
}

function AccountTiledList({accounts}: Props) {

  return (
    <Grid container direction="row" spacing={2}>
      {accounts?.length > 0 ?
        accounts?.map((account: Account) =>
          <AccountTileItem key={account.id}
                           account={account}
          />
        ) :
        <div>You currently do not have any accounts. Feel free to add them now!</div>
      }
    </Grid>
  )
}

export default AccountTiledList;