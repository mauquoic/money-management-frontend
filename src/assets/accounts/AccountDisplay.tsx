import React, {useEffect, useState} from "react";
import AccountTiles from "./AccountTiles";
import AccountList from "./AccountList";
import {Account} from "../../models/Account";
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import AccountService from "./AccountService";
import Grid from "@mui/material/Grid";
import ToggleButtonGroup from "@mui/material/ToggleButtonGroup";
import ToggleButton from "@mui/material/ToggleButton";
import Button from "@mui/material/Button";

function AccountDisplay() {
  const [view, setView] = useState("tiled");
  const [accounts, setAccounts] = useState<Array<Account>>([]);

  const handleView = (event: React.MouseEvent<HTMLElement>, view: string) => {
    setView(view);
  };

  useEffect(() => {
    AccountService.getAccounts().then((accounts: Array<Account>) => setAccounts(accounts));
  }, []);

  return (
    <Grid container>
      <Grid container justifyContent="flex-end">
        <ToggleButtonGroup
          exclusive
          color="primary"
          aria-label="contained button group"
          value={view}
          onChange={handleView}
        >
          <ToggleButton value="list" aria-label="left aligned">
            <FontAwesomeIcon color="black" icon={['fas', 'list']}/>
          </ToggleButton>
          <ToggleButton value="tiled" aria-label="left aligned">
            <FontAwesomeIcon color="black" icon={['fas', 'th']}/>
          </ToggleButton>
        </ToggleButtonGroup>
      </Grid>
      <Grid item xs={12}>
        {view === "tiled" ?
          <AccountTiles accounts={accounts}/> :
          <AccountList accounts={accounts}/>
        }
      </Grid>
      <Grid container justifyContent="flex-end">
        <Button size="small">Add account</Button>
      </Grid>
    </Grid>
  )
};

export default AccountDisplay;