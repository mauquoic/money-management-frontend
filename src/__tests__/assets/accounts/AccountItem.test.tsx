import React from "react";
import {AccountItem} from "../../../assets/accounts/AccountItem";
import {cleanup, render} from "@testing-library/react";

const account = {
  "id": 1,
  "name": "myAccount",
  "description": "This is a description.",
  "currency": "CHF",
  "amount": 100
};

afterEach(cleanup);

describe("AccountItem rendering", () => {
  test("AccountItem is rendered correctly", () => {
    const accountItem = render(<AccountItem account={account} />);
    expect(accountItem).toMatchSnapshot();
  });
});