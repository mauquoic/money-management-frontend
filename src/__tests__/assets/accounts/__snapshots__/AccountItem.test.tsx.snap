// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountItem rendering AccountItem is rendered correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <tr
        class="MuiTableRow-root MuiTableRow-hover css-1q1u3t4-MuiTableRow-root"
        role="checkbox"
        tabindex="-1"
      >
        <td
          class="MuiTableCell-root MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1ce3qcx-MuiTableCell-root"
        >
          myAccount
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1ce3qcx-MuiTableCell-root"
        >
          CHF
           
          100
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1ce3qcx-MuiTableCell-root"
        >
          This is a description.
        </td>
      </tr>
    </div>
  </body>,
  "container": <div>
    <tr
      class="MuiTableRow-root MuiTableRow-hover css-1q1u3t4-MuiTableRow-root"
      role="checkbox"
      tabindex="-1"
    >
      <td
        class="MuiTableCell-root MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1ce3qcx-MuiTableCell-root"
      >
        myAccount
      </td>
      <td
        class="MuiTableCell-root MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1ce3qcx-MuiTableCell-root"
      >
        CHF
         
        100
      </td>
      <td
        class="MuiTableCell-root MuiTableCell-alignCenter MuiTableCell-sizeMedium css-1ce3qcx-MuiTableCell-root"
      >
        This is a description.
      </td>
    </tr>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
